compiler/rustc_mir/src/interpret/intrinsics.rs

```diff
@@ -56,8 +56,12 @@ crate fn eval_nullary_intrinsic<'tcx>(
             let alloc = type_name::alloc_type_name(tcx, tp_ty);
             ConstValue::Slice { data: alloc, start: 0, end: alloc.len() }
         }
-        sym::needs_drop => ConstValue::from_bool(tp_ty.needs_drop(tcx, param_env)),
+        sym::needs_drop => {
+            ensure_monomorphic_enough(tcx, tp_ty)?;
+            ConstValue::from_bool(tp_ty.needs_drop(tcx, param_env))
```

Comment:
Should `ty.needs_drop()` return `true` instead of `false` when the type is not monomorphic enough? It is allowed to return `true` even if dropping is not necessary. This is not allowed the other way around.

Reply:
> Should ty.needs_drop() return true instead of false when the type is not monomorphic enough?

I'm pretty sure it does.
But that might be an overapproximation, so it is still wrong to return here.